torch/ao/quantization/_pt2e/decompositions.py

```diff
@@ -27,3 +27,56 @@ def dequantize_per_tensor(
     dtype: torch.dtype,
 ) -> torch.Tensor:
     return (input.to(torch.float32) - zero_point) * scale
+
+# Helper function used to implement per-channel quantization against any axis
+def _permute_to_axis_zero(x, axis):
+    new_axis_list = list(range(x.dim()))
+    new_axis_list[axis] = 0
+    new_axis_list[0] = axis
+    y = x.permute(tuple(new_axis_list))
+    return y, new_axis_list
+
+@register_decomposition(torch.ops.quantized_decomposed.quantize_per_channel)
+def quantize_per_channel(
+        input: torch.Tensor,
+        scales: torch.Tensor,
+        zero_points: torch.Tensor,
+        axis: int,
+        quant_min: int,
+        quant_max: int,
+        dtype: torch.dtype
+) -> torch.Tensor:
+    input, permute_axis_list = _permute_to_axis_zero(input, axis)
```

Comment:
is it same as calling here input = input.transpose(axis, 0) and then call this again at the end of the function?

Reply:
So btw if this is instead dome as input.transpose(axis, -1) or even input.movedim(axis, -1) (and then the reverse at the end of the function), then regular broadcasting will work

Reply:
yeah feels like this is the same, I can add this as a follow up PR to improve the implementation